flask/json/tag.py

```diff
@@ -0,0 +1,253 @@
+from base64 import b64decode, b64encode
+from datetime import datetime
+from uuid import UUID
+
+from jinja2 import Markup
+from werkzeug.http import http_date, parse_date
+
+from flask._compat import iteritems, text_type
+from flask.json import dumps, loads
+
+
+class JSONTag(object):
+    """Base class for defining type tags for :class:`TaggedJSONSerializer`."""
+
+    __slots__ = ('serializer',)
+    key = None
+    """The tag to mark the serialized object with. If ``None``, this tag is
```

Comment:
Sphinx and PyCharm both recognize it. Not a big deal though, I can switch to `#:` and ignore PyCharm.